fix: align spec with OpenID4VP (compatibility + interoperability) (#840)
italia/eidas-it-wallet-docs: docs/en/remote-flow.rst

          "request_object_signing_alg_values_supported": [
            "ES256"
          ],

>           "client_id_prefixes_supported": ["openid_federation"],

Author: ```suggestion
          "client_id_prefixes_supported": ["openid_federation", "x509_hash"],
```

  For security reasons and to prevent endpoint mix-up attacks, the value contained in the ``request_uri`` parameter MUST be one of those attested by a trusted third party, such as those provided in the ``openid_credential_verifier`` metadata within the ``request_uris`` parameter, obtained from the Trust Chain about the Relying Party (:ref:`WP_081 <wallet-credential-presentation-testcases>` and :ref:`RPR-85 <test-plans-remote-presentation>`).

> .. note::

Author: I suppose that it is removed because it is moved below in the text

     - REQUIRED. Object with Credential format identifiers. See `OpenID4VP`_ Appendix B.
   * - `alg_values_supported`
     - OPTIONAL. Array of cryptographic suites supported. See `OpenID4VP`_ Appendix B.
   * - `client_id_prefixes_supported`

>      - RECOMMENDED. A non-empty array of the Client Identifier Prefixes that the Wallet Instance supports.  Valid values include ``openid_federation`` and ``x509_hash``; if omitted, the default is ``pre-registered``.

Author: we need an additional note for legacy RP using https URIs only, these would implicitly use the openid-federation client id scheme. The Wallet instances interacting with them would evaluate the trust with them collecting the trust chain about them, making them as pre-registered.

    - It MUST be set to ``direct_post.jwt`` (:ref:`RPR-106 <test-plans-remote-presentation>`).
  * - **dcql_query**
    - Object representing a request for a presentation of Credentials, according to the DCQL query language defined in Section 6 of `OpenID4VP`_.
  * - **transaction_data**
    - An optional non-empty array of JSON objects, each describing a transaction that the Relying Party requests the User to authorize. Each transaction object includes: 
        - **type**.  String that identifies the transaction data type.
        - **credential_ids**. Array referencing one or more Credentials from the ``dcql_query`` that can authorize the transaction.  
  * - **transaction_data_hashes_alg** 

>     - An optional array of of strings, each representing a hash algorithm identifier, corresponding to a hash algorithm name listed in the `IANA <https://www.iana.org/assignments/named-information/named-information.xhtml#hash-alg>`_.  One of these algorithms MUST be used to calculate the hashes in the ``transaction_data_hashes`` response parameter.  If omitted, the default hash algorithm is sha-256.

Author: ```suggestion
    - An optional array of of strings, each representing a hash algorithm identifier, corresponding to a hash algorithm name listed in the `IANA <https://www.iana.org/assignments/named-information/named-information.xhtml#hash-alg>`_.  One of these algorithms MUST be used to calculate the hashes in the ``transaction_data_hashes`` response parameter.  If omitted, the default hash algorithm is ``sha-256``.
```

   * - **Error Code**
     - **Description**
   * - ``invalid_request_uri``
     - The `request_uri` in the authorization request returns an error, contains invalid data, or is otherwise malformed. :rfc:`9101`
   * - ``vp_formats_not_supported``
     - The Wallet Instance does not support any of the vp formats required by the Relying Party. `OpenID4VP`_
   * - ``invalid_request_uri_method``
     - The value of the ``request_uri_method`` parameter is neither ``get`` nor ``post``. `OpenID4VP`_
   * - ``invalid_request``
     - The request is malformed or inconsistent, the Client Identifier Prefix is unsupported, or requirements of a prefix are violated. `OpenID4VP`_
   * - ``access_denied``
     - The Wallet did not have the requested credential, the User did not consent, or the Wallet failed to authenticate the User. `OpenID4VP`_
   * - ``invalid_client``
     - The Relying Party cannot be authorized due to trust validation failures or is not a valid participant of the federation. `OID-FED`_
   * - ``invalid_transaction_data``

>      - One or more objects in the ``transaction_data`` structure are invalid — for example, containing unknown or unsupported types, malformed or missing fields, invalid values, or references to unavailable Credentials. `OpenID4VP`_

Author: ```suggestion
     - One or more objects in the ``transaction_data`` structure are invalid. For instance, those objects contain unknown or unsupported types, malformed or missing fields, invalid values, or references to unavailable Credentials. `OpenID4VP`_
```

   * - ``vp_formats_not_supported``
     - The Wallet Instance does not support any of the vp formats required by the Relying Party. `OpenID4VP`_
   * - ``invalid_request_uri_method``
     - The value of the ``request_uri_method`` parameter is neither ``get`` nor ``post``. `OpenID4VP`_
   * - ``invalid_request``
     - The request is malformed or inconsistent, the Client Identifier Prefix is unsupported, or requirements of a prefix are violated. `OpenID4VP`_
   * - ``access_denied``
     - The Wallet did not have the requested credential, the User did not consent, or the Wallet failed to authenticate the User. `OpenID4VP`_
   * - ``invalid_client``
     - The Relying Party cannot be authorized due to trust validation failures or is not a valid participant of the federation. `OID-FED`_
   * - ``invalid_transaction_data``
     - One or more objects in the ``transaction_data`` structure are invalid. For instance, those objects contain unknown or unsupported types, malformed or missing fields, invalid values, or references to unavailable Credentials. `OpenID4VP`_

Relying Party Response
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> As defined in Section 8.2. (Response Mode "direct_post") of the `OpenID4VP`_ specification, if the Response URI has successfully processed the Authorization Response or Authorization Error Response, it MUST respond with an HTTP status code of 200 with ``Content-Type`` of ``application/json`` and a JSON object in the response body.

Author: ```suggestion
As defined in Section 8.2. (Response Mode ``direct_post``) of the `OpenID4VP`_ specification, if the Response URI has successfully processed the Authorization Response or Authorization Error Response, it MUST respond with an HTTP status code of 200 with ``Content-Type`` of ``application/json`` and a JSON object in the response body.
```
